src/figuresClasses.ts

```diff
@@ -1,19 +1,81 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
 
+type Color = 'red' | 'green' | 'blue';
+export interface Figure {
+  color: Color;
+  shape?: Shape;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  constructor(
+    public color: Color,
+    public sideA: number,
+    public sideB: number,
+    public sideC: number,
+    public shape?: Shape,
+  ) {
+    this.shape = 'triangle';
 
-}
+    if (sideA <= 0 || sideA <= 0 || sideC <= 0) {
+      throw new Error('Incorrect values for triangle sides');
+    }
+
+    const maxSide: number = Math.max(sideA, Math.max(sideB, sideC));
+
+    if (maxSide >= (sideA + sideB + sideC - maxSide)) {
+      throw new Error('It is not a triangle');
+    }
+  }
 
-export class Circle {
+  getArea(): number {
+    const sunSides: number = this.sideA + this.sideB + this.sideC;
+    const semiPerimeter: number = sunSides / 2;
 
+    return Math.floor(Math.sqrt(semiPerimeter
+      * (semiPerimeter - this.sideA)
+      * (semiPerimeter - this.sideB)
+      * (semiPerimeter - this.sideC)) * 100) / 100;
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  constructor(
+    public color: Color,
+    public radius: number,
+    public shape?: Shape,
+  ) {
+    this.shape = 'circle';
 
+    if (radius <= 0) {
+      throw new Error('Incorrect values for circle radius');
+    }
+  }
+
+  getArea(): number {
+    return Math.floor((Math.PI * this.radius ** 2) * 100) / 100;
+  }
 }
 
-export function getInfo(figure) {
+export class Rectangle implements Figure {
+  constructor(
+    public color: Color,
+    public width: number,
+    public height: number,
+    public shape?: Shape,
+  ) {
+    this.shape = 'rectangle';
```

Comment:
we can use this implementetion in class body, because all rectangles will have this propery

Reply:
done